src/main/java/com/artipie/auth/AuthFromYaml.java

```diff
@@ -74,24 +68,13 @@ public AuthFromYaml(final YamlMapping cred) {
     public Optional<Authentication.User> user(final String user, final String pass) {
         final YamlMapping users = this.cred.yamlMapping("credentials");
         Optional<Authentication.User> res = Optional.empty();
-        //@checkstyle NestedIfDepthCheck (10 lines)
         if (users != null && users.yamlMapping(user) != null) {
             final String stored = users.yamlMapping(user).string("pass");
-            if (stored != null) {
-                final String type = users.yamlMapping(user).string("type");
-                if (type != null) {
-                    if (check(stored, type, pass)) {
-                        res = Optional.of(
-                            new User(user, AuthFromYaml.groups(users.yamlMapping(user)))
-                        );
-                    }
-                } else {
-                    if (check(stored, pass)) {
-                        res = Optional.of(
-                            new User(user, AuthFromYaml.groups(users.yamlMapping(user)))
-                        );
-                    }
-                }
+            final String type = users.yamlMapping(user).string("type");
```

Comment:
@genryxy passing nullable `type` into `check` method does not seem elegant, and also method with name `check` does not suppose to extract password type... So, I'd suggest create separate method to obtain password type (simply extract this logic from `check`) and then call `check` method with known password type.

Reply:
@olenagerasimova thanks. If I obtain password type from the method it becomes more difficult to distinguish password that have type at the beginning `plain:123` from the password `123`. I can pass `user` and `users` to `check()` method instead of nullable `type` but in this case method will have 4 parameters. Or would it be better to find a solution that doesn't pass a value to the method?

Reply:
@genryxy I see, sorry for the misguided suggestion. Maybe we can make `type` `Optional<String>` to avoid null and rename `check` method properly? The name of the method should show shortly explain what the method does.